### PR TITLE
invoice: swap RouteHop for RouteHint

### PR DESCRIPTION
**Pull request by valentinewallace**
To prevent naming conflicts in bindings

### CODE REVIEW

File: lightning-invoice/src/de.rs [hunk 1]
```diff
@@ -585,12 +587,16 @@ impl FromBase32 for Route {
...
 
-			let hop = RouteHop {
-				pubkey: PublicKey::from_slice(&hop_bytes[0..33])?,
-				short_channel_id: channel_id,
-				fee_base_msat: parse_int_be(&hop_bytes[41..45], 256).expect("slice too big?"),
-				fee_proportional_millionths: parse_int_be(&hop_bytes[45..49], 256).expect("slice too big?"),
-				cltv_expiry_delta: parse_int_be(&hop_bytes[49..51], 256).expect("slice too big?")
+			let hop = RouteHint {
+				src_node_id: PublicKey::from_slice(&hop_bytes[0..33])?,
+				short_channel_id: u64::from_be_bytes(channel_id),
```
Comment (TheBlueMatt) [inline lightning-invoice/src/de.rs:592]: I think you can just swap this for `parse_int_be` like the other integer fields below.

**Review by TheBlueMatt — COMMENTED** [review]
Looks good at first glance.

**Comment by TheBlueMatt** [comment]
Hmm, I wonder if it actually makes sense to further refactor a bit - a `RouteHint` isn't really itself a hint, but one hop in a potentially-but-rarely-larger `RouteHint`. The `lightning_invoice::Route` would then be a `RouteHint` making `lightning::routing::router::RouteHint` a `RouteHintHop` (though presumably we'd put both types in `lightning::routing::router` and use them consistently).

**Comment by valentinewallace** [comment]
> Hmm, I wonder if it actually makes sense to further refactor a bit - a `RouteHint` isn't really itself a hint, but one hop in a potentially-but-rarely-larger `RouteHint`. The `lightning_invoice::Route` would then be a `RouteHint` making `lightning::routing::router::RouteHint` a `RouteHintHop` (though presumably we'd put both types in `lightning::routing::router` and use them consistently).

Hum so moving the invoice crate's `RouteHint` (previously named `Route`) is a little involved. OK to put off?

File: lightning-invoice/src/ser.rs [hunk 2]
```diff
@@ -365,22 +365,22 @@ impl Base32Len for Fallback {
...
-impl ToBase32 for Route {
+impl ToBase32 for RouteHint {
 	fn write_base32<W: WriteBase32>(&self, writer: &mut W) -> Result<(), <W as WriteBase32>::Err> {
 		let mut converter = BytesToBase32::new(writer);
 
 		for hop in self.iter() {
-			converter.append(&hop.pubkey.serialize()[..])?;
-			converter.append(&hop.short_channel_id[..])?;
+			converter.append(&hop.src_node_id.serialize()[..])?;
+			converter.append(&hop.short_channel_id.to_be_bytes())?;
```
Comment (TheBlueMatt) [inline lightning-invoice/src/ser.rs:374]: Oops, I think this (and the from_be_bytes calls) bumps lightning-invoice's MSRV - we should change the CI to also build+test lightning-invoice on 1.30 not just lightning.

**Comment by TheBlueMatt** [comment]
> Hum so moving the invoice crate's RouteHint (previously named Route) is a little involved. OK to put off?

Hmm, up to you, is the issue the trait implementations? It should still be pretty practical, but its no huge rush.

Otherwise looks good.

**Review by jkczyz — APPROVED** [review]

**Review by TheBlueMatt — APPROVED** [review]

**Pull request merged**